File: express/src/service/SeriesService.ts
import { Series } from '../entity/Series';

export interface iSeriesService {
  findByPageAndSizeAndFilterAndOrder(page: number, size: number, filter?: string, order?: string, ascendingDirection?: boolean): Promise<[Series[], number]>
  findOne(id: number): Promise<Series>;
  save(entity: Series): Promise<Series>;
  update(id: number, entity: Series): Promise<boolean>;
}

File: spring/src/main/java/hu/marko/szakdolgozat/spring/service/SeriesService.java
package hu.marko.szakdolgozat.spring.service;

import hu.marko.szakdolgozat.spring.service.model.PageModel;
import hu.marko.szakdolgozat.spring.service.model.Series;
import hu.marko.szakdolgozat.spring.service.model.UpdateSeries;

public interface SeriesService {
  PageModel<Series> findByPageAndSizeAndFilterAndOrder(Integer page, Integer size, String filter, String order,
      Boolean ascendingDirection);

  Series findOne(Long id);

  Series save(Series series);

  Boolean update(Long id, UpdateSeries entity);

  Boolean deleteImage(Long seriesId);
}
